lib/runners/java.js

```diff
@@ -1,86 +1,197 @@
 var shovel = require('../shovel'),
-    exec = require('child_process').exec,
-    util = require('../util');
+    util = require('../util'),
+    tagHelpers = require('../utils/tag-helpers'),
+    manipulateFileSync = require('../utils/manipulate-file-sync'),
+    fs = require('fs-extra');
 
 module.exports.run = function run(opts, cb) {
+  // we will cache the contents of this file so that we can process it later.
+  let buildGradle = '';
 
   shovel.start(opts, cb, {
     solutionOnly: function(runCode, fail) {
+      fs.emptydirSync(`${opts.dir}/src`);
+
       const names = extractNames(opts.solution, 'Solution');
-      const solutionFile = util.codeWriteSync('javac', opts.solution, opts.dir, names.file);
-
-      compile([solutionFile], function(error, stdout, stderr) {
-        if (error) return fail(error, stdout, stderr);
-        opts.publish('stdout', stdout);
-        runCode({
-          name: 'groovyclient',
-          args: [
-            '-cp', opts.dir,
-            '-e', `import ${names.full}; ${names.full}.main(null);`,
-            '-Cauthtoken', 'groovy'
-          ]
-        });
-      });
+      util.codeWriteSync(null, opts.solution, `${opts.dir}/src/main/java`, names.file);
```

Comment:
This is just
```js
util.writeFileSync(`${opts.dir}/src/main/java`, names.file, opts.solution, true);
```
because language parameter for `util.codeWriteSync` is `null`. `!config.moduleRegExs[null]` so `defaultFileName` is returned.

Also if we're using `fs-extra`, there is [`fs.outputFileSync`](https://github.com/jprichardson/node-fs-extra/blob/master/docs/outputFile-sync.md) that creates the directory if it doesn't exist already. So this becomes:
```js
fs.outputFileSync(`${opts.dir}/src/main/java/${names.file}`, opts.solution).
```

---

By the way, I never understood why `util.codeWriteSync` or shared `config.moduleRegExs` exists. Is there a reason to have those definitions in a shared module?